app/controllers/users_controller.rb

```diff
@@ -1490,28 +1498,34 @@ def confirm_admin
     end
   end
 
+  def confirm_session
+    # TODO(pmusaraj): add support for confirming via passkey, 2FA
+    params.require(:password)
+
+    if SiteSetting.enable_discourse_connect || !SiteSetting.enable_local_logins
+      raise Discourse::NotFound
+    end
+
+    if confirm_secure_session
+      render json: success_json
+    else
+      render json: failed_json.merge(error: I18n.t("login.incorrect_password"))
+    end
+  end
+
+  def trusted_session
```

Comment:
I have the same comment as https://github.com/discourse/discourse/pull/23587/files#r1351136186 as well. Why does `secure_session_confirmed?` matter in the context of passkey which is passwordless?